### ISSUE TITLE
Tutorial on chemical properties using SMILE

### ISSUE THREAD

**Issue by crockpotveggies** [deeplearning4j/deeplearning4j-docs#29]
## Due Date
*To be completed by:* 2018-04-25

## Description
@jrmerwin As discussed, forming a deep learning tutorial on how to predict a chemical property when using its composition in a SMILE format.

## Assignees
*Please ensure you have assigned at least one person to this issue. Include any authors and reviewers required.*

**Comment by crockpotveggies** [comment]
50 million ~ 1 billion compounds in SMILE format: http://gdb.unibe.ch/downloads/

**Comment by crockpotveggies** [comment]
20 million compounds, perhaps also useful: https://github.com/isayev/ANI1_dataset

**Comment by turambar** [comment]
Did we consult @AlexDBlack on this? He likely has strong opinions about it, some war stories about the SMILES data, and *maybe* some code we can repurpose?

**Comment by AlexDBlack** [comment]
Yeah, @crockpotveggies and I discussed this privately on gitter.
tl;dr data in just about any format would work for a basic demo, as we can convert it to SMILES or whatever as required.
As for the net, I was suggesting perhaps an MLP using ECFP4, and maybe (depending on the task) also a bidirectional RNN directly on SMILES. I'm happy to do the ECFP4 conversion if required to produce a 'clean' dataset.
If we go the RNN route, it might make sense to finally build a proper character sequence record reader in DataVec.
